lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -403,6 +403,32 @@ static bool diagnoseOperatorJuxtaposition(UnresolvedDeclRefExpr *UDRE,
   }
 }
 
+
+static bool diagnoseRangeOperatorMisspell(UnresolvedDeclRefExpr *UDRE,
+                                          TypeChecker &TC) {
+  auto name = UDRE->getName().getBaseIdentifier();
+  if (!name.isOperator())
+    return false;
+
+  auto corrected = StringRef();
+  if (name.str() == ".." || name.str() == "...." ||
```

Comment:
Maybe there possible a more general solution here instead of hard-coding some of the misspellings directly?

Reply:
Have you read the header? 
I could stick it into `performTypoCorrection` if I manually do an unqualified lookup for `Comparable` and include it's members, but it sounds way too expensive compared to the hard-coding approach.

Reply:
Sorry, I mean besides looking up stuff on `Comparable`? It seems like if we already know that something is a binary operator we could just lookup all global ones and filter based on closeness? I think we should already have code to retrieve operators like that...

Reply:
Range operators are a `Comparable` extension (not global). We do have a method that looks up all visible operators in code completion, but nothing is filtered based on closeness there. Apparently that's what we have `performTypoCorrection` for, but it doesn't look up any member decls from other modules. I assume not to waste time, which we would be doing by performing any kind of look up that finds the range operators..